bst.py

```diff
@@ -40,37 +41,165 @@ def size(self):
+        left = self.nodes[node].get('left')
+        right = self.nodes[node].get('right')
+        if left is not None:
+            yield "\t%s -> %s;" % (node, left)
+            for i in self._get_dot(left):
+                yield i
+        elif right is not None:
+            r = random.randint(0, 1e9)
+            yield "\tnull%s [shape=point];" % r
+            yield "\t%s -> null%s;" % (node, r)
+        if right is not None:
+            yield "\t%s -> %s;" % (node, right)
+            for i in self._get_dot(right):
+                yield i
+        elif left is not None:
+            r = random.randint(0, 1e9)
+            yield "\tnull%s [shape=point];" % r
+            yield "\t%s -> null%s;" % (node, r)
+
+    def print_dot(self):
+        # To run me do this from cmd line: python bst.py > temp.png
+        import subprocess
+        dot_graph = self.get_dot()
+        t = subprocess.Popen(["dot", "-Tpng"], stdin=subprocess.PIPE)
+        t.communicate(dot_graph)
+
+    def _in_order(self, node):
+        '''Helper function for in_order method.'''
+        left = self.nodes[node].get('left')
+        right = self.nodes[node].get('right')
+
+        if left:
+            for elem in self._in_order(left):
+                yield elem
+        yield node
+        if right:
+            for elem in self._in_order(right):
+                yield elem
+
+    def in_order(self):
+        '''Return the values in the tree using in-order traversal.'''
+        for elem in self._in_order(self.start):
+            yield elem
+
+    def _pre_order(self, node):
+        '''Helper function for pre_order method.'''
+        left = self.nodes[node].get('left')
+        right = self.nodes[node].get('right')
+
+        yield node
+        if left:
+            for elem in self._pre_order(left):
+                yield elem
+        if right:
+            for elem in self._pre_order(right):
+                yield elem
+
+    def pre_order(self):
+        '''Return the values in the tree using pre-order traversal.'''
+        for elem in self._pre_order(self.start):
+            yield elem
+
+    def _post_order(self, node):
+        '''Helper function for post_order method.'''
+        left = self.nodes[node].get('left')
+        right = self.nodes[node].get('right')
+
+        if left:
+            for elem in self._post_order(left):
+                yield elem
+        if right:
+            for elem in self._post_order(right):
+                yield elem
+        yield node
+
+    def post_order(self):
+        '''Return the values in the tree using post-order traversal.'''
+        for elem in self._post_order(self.start):
+            yield elem
+
+    def breadth_first(self):
+        '''Return the values in the tree using breadth-first traversal.'''
+        visited = []
+        if self.start is None:
+            return
+        # yield self.start
+        # visited.append(self.start)
+        start_visited = visited
```

Comment:
this is now a second pointer to the same empty list.  Nothing is going to happen.